lib/Client.php

```diff
@@ -353,9 +332,29 @@ protected function commitSession() {
 	 * @throws OpenIDConnectClientException
 	 */
 	protected function fetchURL($url, $post_body = null, $headers = []) {
-		// TODO: see how to use ownCloud HttpClient ....
+		$this->logger->debug("Fetching URL: $url");
 		try {
-			return parent::fetchURL($url, $post_body, $headers);
+			$client = $this->clientService->newClient();
+			if ($post_body === null) {
+				return $client->get($url, [
+					'headers' => $headers,
+				])->getBody();
+			}
+			// Default content type is form encoded
+			$content_type = 'application/x-www-form-urlencoded';
+
+			// Determine if this is a JSON payload and add the appropriate content type
+			if (\is_object(json_decode($post_body))) {
+				$content_type = 'application/json';
+			}
```

Comment:
The caller should handle decoding, see: https://github.com/owncloud/openidconnect/blob/master/lib/Client.php#L156

This should only fetch.

Reply:
Nice to have for support: In case it's not json, dump the first ~300 Lines of the body in to the error-log so we can instantly detect that we are getting some proxy error page. ;)

Reply:
the code you are referring to is for the request body - not the response body ....